### PR TITLE
Parallelize execution of rolling updates

### PR DESCRIPTION
**Pull request by rohansingh**
Have a thread pool of 5 threads which we use to run the rolling update step
for each deployment group. Hopefully this speeds things up a bit.

### CODE REVIEW

**Comment by davidxia** [comment]
assuming tests pass, :+1:

**Comment by rohansingh** [comment]
@gimaker Want to take a look as well? I think you know Java stuff better than either of us.

File: helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUpdateService.java [hunk 1]
```diff
@@ -132,12 +145,17 @@ public void run(final boolean timeout) throws InterruptedException {
     @Override
     public void run(final boolean timeout) throws InterruptedException {
       for (final DeploymentGroup dg : masterModel.getDeploymentGroups().values()) {
-        try {
-          masterModel.rollingUpdateStep(dg, DefaultRolloutPlanner.of(dg));
-        } catch (Exception e) {
-          log.warn("error processing rolling update step for deployment group: {} - {}",
-                   dg.getName(), e);
-        }
+        rollingUpdateExecutor.execute(new Runnable() {
```
Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUpdateService.java]: probably very much a non-issue but this looks like an unbounded queue

File: helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUpdateService.java [hunk 2]
```diff
@@ -60,6 +65,14 @@
   private final Reactor hostUpdateReactor;
   private final Reactor rollingUpdateReactor;
 
+  private final ExecutorService rollingUpdateExecutor = MoreExecutors.getExitingExecutorService(
```
Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUpdateService.java]: The exiting executor service seems unnecessary

**Pull request merged**